### ISSUE TITLE
VRNN trained with padded values

### ISSUE THREAD

**Issue by ArthurD1** [dnguyengithub/MultitaskAIS#22]
Hello,

First thank you for sharing your code, your paper is very interesting.
I went through your code and I could not seem to find where you deal with padded values.
I feel like the VRNN is being trained with these values so a bunch of tensor of size data_dim filled with 0. Sometimes in the same batch you may have some very large differences in term of sequence sizes, from 20 to 144 for instance.

Did I miss something or do we consider them as noisy data the VRNN has to deal with? 
Thank you

**Comment by duonginspace** [comment]
Hi,

> I could not seem to find where you deal with padded values.
Sorry I didnt understand your question, could you rephrase it?

> Sometimes in the same batch you may have some very large differences in term of sequence sizes, from 20 to 144 for instance.
VRNN, whose the backbone is a LSTM, can naturally deal with variable-length sequences. When we calculate the total loss, the lower bound at each timestep is multiplied by the cur_mask.

Hope that this answers your question.
